BSP/Inc/BSP_SPI.h

```diff
@@ -11,6 +11,13 @@
 #include "common.h"
 #include "bsp.h"
 
+typedef struct _spi_message {
+    int8_t opcode;
+    int8_t requestedPort;
+    int8_t data;
+} spi_message;
```

Comment:
It's not immediately obvious to me what these fields are, so maybe better names are in order. Is the first field referring to read/write? If so, maybe a boolean value `write` would be better. The `requestedPort` may be better as an enum value, rather than a plain `int8_t`. Maybe a name like `cs` (for chip select) would be more in line with typical jargon.

Will we ever have to worry about having multiple data bytes in a single frame? If so, then we may want to provide a `uint8_t *data_buf` rather than a plain integer.

Reply:
Ok I see now that this refers to an operation and register selection specifically for the chip that we are using, which makes more sense to me now. Just caught me off guard to see it in the BSP file rather than a driver layer file.